src/dnsmasq_interface.c

```diff
@@ -820,29 +821,98 @@ void _FTL_iface(struct irec *recviface, const char *file, const int line)
 	next_iface.name[0] = '-';
 	next_iface.name[1] = '\0';
 
+	// Check if we need to identify the receving interface by its address
+	if(!recviface && addr &&
+	   ((addrfamily == AF_INET && addr->addr4.s_addr != 0) ||
+	    (addrfamily == AF_INET6 && addr->addr6.s6_addr[0] != 0)))
+	{
+		if(config.debug & DEBUG_NETWORKING)
+		{
+			char addrstr[INET6_ADDRSTRLEN] = { 0 };
+			if(addrfamily == AF_INET)
+				inet_ntop(AF_INET, &addr->addr4, addrstr, INET6_ADDRSTRLEN);
+			else // if(addrfamily == AF_INET6)
+				inet_ntop(AF_INET6, &addr->addr6, addrstr, INET6_ADDRSTRLEN);
+			logg("Identifying interface (looking for %s):", addrstr);
```

Comment:
This part is only executed while debug flags are set. But more than logging happens. Should it be like that?